bot/exts/utils/thread_bumper.py

```diff
@@ -0,0 +1,142 @@
+                await self.threads_to_bump.delete(thread_id)
+                continue
+
+            if thread.archived:
+                threads_to_maybe_bump.append(thread)
+
+        await self.unarchive_threads_not_manually_archived(threads_to_maybe_bump)
+
+    @commands.group(name="bump")
+    async def thread_bump_group(self, ctx: commands.Context) -> None:
+        """A group of commands to manage the bumping of threads."""
+        if not ctx.invoked_subcommand:
+            await ctx.send_help(ctx.command)
+
+    @thread_bump_group.command(name="add")
+    async def add_thread_to_bump_list(self, ctx: commands.Context, thread: t.Optional[discord.Thread]) -> None:
+        """Add a thread to the bump list."""
+        await self.init_task
+
+        if not thread:
+            if isinstance(ctx.channel, discord.Thread):
+                thread = ctx.channel
+            else:
+                raise commands.BadArgument("You must provide a thread, or run this command within a thread.")
+
+        await self.threads_to_bump.set(thread.id, "sentinel")
+        await ctx.send(f":ok_hand:{thread.mention} has been added to the bump list.")
+
+    @thread_bump_group.command(name="remove", aliases=("r", "rem", "d", "del", "delete"))
+    async def remove_thread_from_bump_list(self, ctx: commands.Context, thread: t.Optional[discord.Thread]) -> None:
+        """Remove a thread from the bump list."""
+        await self.init_task
+
+        if not thread:
+            if isinstance(ctx.channel, discord.Thread):
+                thread = ctx.channel
+            else:
+                raise commands.BadArgument("You must provide a thread, or run this command within a thread.")
+
+        await self.threads_to_bump.delete(thread.id)
+        await ctx.send(f":ok_hand: {thread.mention} has been removed from the bump list.")
+
+    @thread_bump_group.command(name="list", aliases=("get",))
+    async def list_all_threads_in_bump_list(self, ctx: commands.Context) -> None:
+        """List all the threads in the bump list."""
+        await self.init_task
+
+        lines = [f"<#{k}>" for k, _ in await self.threads_to_bump.items()]
+        embed = discord.Embed(
+            title="Threads in the bump list",
+            colour=constants.Colours.blue
+        )
+        await LinePaginator.paginate(lines, ctx, embed)
+
+    @commands.Cog.listener()
+    async def on_thread_update(self, _: discord.Thread, after: discord.Thread) -> None:
+        """
+        Listen for thread updates and check if the thread has been archived.
+
+        If the thread has been archived, and is in the bump list, un-archive it.
+        """
+        await self.init_task
+
+        if not after.archived:
+            return
+
+        bumped_threads = [k for k, _ in await self.threads_to_bump.items()]
+        if after.id in bumped_threads:
+            await self.unarchive_threads_not_manually_archived([after])
```

Comment:
PS: I believe you can do something like this here instead.
```suggestion
        if await bumped_threads.contains(after.id):
            await self.unarchive_threads_not_manually_archived([after])
```

Reply:
You could also use [`RedisCache.contains()`](https://github.com/SebastiaanZ/async-rediscache/blob/6a68850e48cc36f0a60a3ddd12bbf3c908e10380/async_rediscache/types/cache.py#L54-L55) in line 88 if you decide to incorporate [this](https://github.com/python-discord/bot/pull/2084/files#r810018069) suggestion.